typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -250,9 +251,17 @@
 ;;  make-predicate
 ;;  cast
 
-;; Helper to construct syntax for contract definitions
+;; Helpers to construct syntax for contract definitions
+;; make-contract-def-rhs : Type-Stx Boolean Boolean -> Syntax
 (define (make-contract-def-rhs type flat? maker?)
-  (contract-def-property #'#f `#s(contract-def ,type ,flat? ,maker? untyped)))
+  (define contract-def `#s(contract-def ,type ,flat? ,maker? untyped))
+  (contract-def-property #'#f (λ () contract-def)))
+
+;; make-contract-def-rhs/from-typed : Id Boolean Boolean -> Syntax
+(define (make-contract-def-rhs/from-typed id flat? maker?)
+  (contract-def-property
+   #'#f
+   (λ () `#s(contract-def ,(or (cast-table-ref id) #'Any) ,flat? ,maker? typed))))
```

Comment:
This was just fixed by @takikawa so that the `Any` isn't necessary after all, even at the top level. What he did was have `cast` just expand to `core-cast` within an `#%expression` form, so that `core-cast` will never actually be at the top level, even if `cast` is.